content/en/docs/Reference/veneers.md

```diff
@@ -0,0 +1,397 @@
+- type: olm.bundle.object
+  value:
+    data: eyJhcGlWZXJzaW9uIjoiYXBpZXh0ZW5zaW9ucy5rOHMuaW8vdjEiLCJraW5kIjoiQ3VzdG9tUmVzb3VyY2VEZWZpbml0aW9uIiwibWV0YWRhdGEiOnsiYW5ub3RhdGlvbnMiOnsiY29udHJvbGxlci1nZW4ua3ViZWJ1aWxkZXIuaW8vdmVyc2lvbiI6InYwLjYuMSJ9LCJjcmVhdGlvblRpbWVzdGFtcCI6bnVsbCwibmFtZSI6ImFwcHMuZXhhbXBsZS5jb20ifSwic3BlYyI6eyJncm91cCI6ImV4YW1wbGUuY29tIiwibmFtZXMiOnsia2luZCI6IkFwcCIsImxpc3RLaW5kIjoiQXBwTGlzdCIsInBsdXJhbCI6ImFwcHMiLCJzaW5ndWxhciI6ImFwcCJ9LCJzY29wZSI6Ik5hbWVzcGFjZWQiLCJ2ZXJzaW9ucyI6W3sibmFtZSI6InYxIiwic2NoZW1hIjp7Im9wZW5BUElWM1NjaGVtYSI6eyJkZXNjcmlwdGlvbiI6IkFwcCBpcyB0aGUgU2NoZW1hIGZvciB0aGUgYXBwcyBBUEkiLCJwcm9wZXJ0aWVzIjp7ImFwaVZlcnNpb24iOnsiZGVzY3JpcHRpb24iOiJBUElWZXJzaW9uIGRlZmluZXMgdGhlIHZlcnNpb25lZCBzY2hlbWEgb2YgdGhpcyByZXByZXNlbnRhdGlvbiBvZiBhbiBvYmplY3QuIFNlcnZlcnMgc2hvdWxkIGNvbnZlcnQgcmVjb2duaXplZCBzY2hlbWFzIHRvIHRoZSBsYXRlc3QgaW50ZXJuYWwgdmFsdWUsIGFuZCBtYXkgcmVqZWN0IHVucmVjb2duaXplZCB2YWx1ZXMuIE1vcmUgaW5mbzogaHR0cHM6Ly9naXQuazhzLmlvL2NvbW11bml0eS9jb250cmlidXRvcnMvZGV2ZWwvc2lnLWFyY2hpdGVjdHVyZS9hcGktY29udmVudGlvbnMubWQjcmVzb3VyY2VzIiwidHlwZSI6InN0cmluZyJ9LCJraW5kIjp7ImRlc2NyaXB0aW9uIjoiS2luZCBpcyBhIHN0cmluZyB2YWx1ZSByZXByZXNlbnRpbmcgdGhlIFJFU1QgcmVzb3VyY2UgdGhpcyBvYmplY3QgcmVwcmVzZW50cy4gU2VydmVycyBtYXkgaW5mZXIgdGhpcyBmcm9tIHRoZSBlbmRwb2ludCB0aGUgY2xpZW50IHN1Ym1pdHMgcmVxdWVzdHMgdG8uIENhbm5vdCBiZSB1cGRhdGVkLiBJbiBDYW1lbENhc2UuIE1vcmUgaW5mbzogaHR0cHM6Ly9naXQuazhzLmlvL2NvbW11bml0eS9jb250cmlidXRvcnMvZGV2ZWwvc2lnLWFyY2hpdGVjdHVyZS9hcGktY29udmVudGlvbnMubWQjdHlwZXMta2luZHMiLCJ0eXBlIjoic3RyaW5nIn0sIm1ldGFkYXRhIjp7InR5cGUiOiJvYmplY3QifSwic3BlYyI6eyJkZXNjcmlwdGlvbiI6IkFwcFNwZWMgZGVmaW5lcyB0aGUgZGVzaXJlZCBzdGF0ZSBvZiBBcHAiLCJwcm9wZXJ0aWVzIjp7ImJhciI6eyJkZXNjcmlwdGlvbiI6IkJhciBpcyBhbiBleGFtcGxlIGZpZWxkIG9mIEFwcC4iLCJ0eXBlIjoic3RyaW5nIn0sImZvbyI6eyJkZXNjcmlwdGlvbiI6IkZvbyBpcyBhbiBleGFtcGxlIGZpZWxkIG9mIEFwcC4iLCJ0eXBlIjoic3RyaW5nIn19LCJ0eXBlIjoib2JqZWN0In0sInN0YXR1cyI6eyJkZXNjcmlwdGlvbiI6IkFwcFN0YXR1cyBkZWZpbmVzIHRoZSBvYnNlcnZlZCBzdGF0ZSBvZiBBcHAiLCJ0eXBlIjoib2JqZWN0In19LCJ0eXBlIjoib2JqZWN0In19LCJzZXJ2ZWQiOnRydWUsInN0b3JhZ2UiOnRydWUsInN1YnJlc291cmNlcyI6eyJzdGF0dXMiOnt9fX1dfSwic3RhdHVzIjp7ImFjY2VwdGVkTmFtZXMiOnsia2luZCI6IiIsInBsdXJhbCI6IiJ9LCJjb25kaXRpb25zIjpbXSwic3RvcmVkVmVyc2lvbnMiOltdfX0=
+relatedImages:
+- image: gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
+  name: ""
+- image: docker.io/example/example-operator-bundle:0.2.0
+  name: ""
+- image: docker.io/example/example-operator:0.2.0
+  name: ""
+schema: olm.bundle
+
+```
+</details>
+
+## Semver Veneer
+
+Since a `veneer` is identified as an input schema which is processed to generate a valid FBC, we can define a `semver veneer` as a schema which uses channel conventions to facilitate the auto-generation of channels adherent to [Semantic Versioning](https://semver.org/) (semver) guidelines. 
```

Comment:
@grokspawn @michaelryanpeter: 

**We do not want auto-generate the channels to facilitate. We want to do that to ensure the quality of the content in the catalogue.** What motivates this feature is to provide consistent data/content across the index and not break cluster upgrades.

So, we can improve here and make more clear why the `Semver Veneer` is best for anyone that is looking for ensuring the quality of the content:

- This option has validations which avoid having upgrade graphs that do not work. That means, avoiding the scenario where misconfiguration done by Operator authors cannot submit solutions that will result in OLM users getting stuck Examples of use cases:
- invalid replaces 
- Inconsistent channels names which have no meaning for cluster admins 
- Ensure that channels and Operators Bundle versions follow up the semver convention which allows cluster admins know exactly what are the versions that then can safely upgrade because are only patch releases and those ones which are versions that they MUST plan beforehand to upgrade because are breaking changes
- Another advantage here is to ensure that the whole catalogue provides the same terminology for the channels which makes it easier for cluster admins to know what is the maturity of the solutions published under the channels instead of each Operator Author choosing a term that might only make sense for them. 
- Encourage and promote standards across the catalog 

Therefore, we should recommend its adoption for those Index Admins which are versioning indexes to allow Operator Authors such as `myregistry/myindex:1.0` and `myregistry/myindex:20`

/hold

I am holding this one because this part IMO is very important. We need really be able to clarify what are the problems that these options address properly here.

Reply:
```suggestion
Since a `veneer` is identified as an input schema which is processed to generate a valid FBC, we can define a `semver veneer` as a schema which uses channel conventions to ensure standards and the quality of the content distribute in the index catalog. Therefore, this veener encourages the adoption of seemver and provides channel auto-generation adherent to [Semantic Versioning](https://semver.org/) (semver) guidelines to facilitate the consumption of what is distributed within and its consumption for cluster admins.  

### When is recommended for index catalog admins to adopt this veener

- If they are looking to ensure that all Operator solutions distributed on the index will follow up semver
- If they are looking to ensure consistent terminology across the index catalogue intend to help cluster admins better and more intuitively understand the maturity level of what is distributed in each channel. 
- If you are versioning the indexes and allowing Operator authors to provide different packages schemas per index version this option can avoid production disruptions such as when cluster admins must manually change the subscription because after a cluster or index upgrades the channel where they are subscribed no longer exist
- If you are looking to facilitate for cluster admins are able to properly know what are the versions that they can safely upgrade because for example just a patch release or when they must better plan these upgrades in production because the next version is a MAJOR bump which contains braking changes and can affect their workloads. 

```

Reply:
I don't agree with you here.  The operator author benefits from the reduced labor.  Our catalog benefits from the consistency.  All the author cares about w.r.t. the catalog is that it's easy to get their operator there, and for users to pull it from there.

Reply:
HI @grokspawn, 

> I don't agree with you here. The operator author benefits from the reduced labor.

You can add it as an advantage but that is not what motivates the veener.  
The motivation for looking for generating the channels is not to make it easier for the Operator authors at all. It is about ensuring consistency for those who will consume the solutions.

We have a huge problem with OCP cluster upgrades because when cluster admins upgrade their cluster they get stuck when the new index version does not exist in the channel where they are subscribed. The same bad user experience can happen if we create an index per k8s versions releases. (It is a very common case scenario that the best practices naming convention try to sort out. we shared it in the f2f as well. )

Who chooses the veener? Are the Operator Authors or the Index Admin? 

Therefore, the index admin will make the choice according to what is best to address their needs in order to make things better for those who consume the catalog itself which is the cluster admin.

Reply:
Only the operator author will interact with veneers.  The rest of the pipeline deals only with FBC.  Therefore, the operator author selects the veneer that best suits them (or forks, implements their own, etc.).
  
I can totally foresee that -- if we do it right -- index owners will come to prefer veneer-generated FBC instead of hand-generated FBC.  I guess that's where I see the benefits of an incentivized opt-in approach (based on award/achievement?), since then index owners can incentivize  a "this tall to ride" approach.  

For example, if the lead time for index inclusion is typically 1 week, the index owner can offer to shorten it to a couple of days if a veneer was used to generate an operator's FBC (due to a higher confidence in the consistency and durability of the resulting FBC). 

I believe that a prescriptive approach will be detrimental to our contributing authors.  I have had a look at the marketplace catalogs, and they are considerably disparate from what we propose,.  Therefore, any required compliance would introduce friction as it imposes costs.  

Since we are in early alpha, we are working to interact with our contributing authors to define the veneers which do the best job of capturing their use-cases.